src/Corrade/Utility/Directory.cpp

```diff
@@ -429,8 +445,15 @@ std::string libraryLocation(const void* address) {
     #elif defined(CORRADE_TARGET_WINDOWS) && !defined(CORRADE_TARGET_WINDOWS_RT)
     HMODULE module{};
     if(!GetModuleHandleExA(GET_MODULE_HANDLE_EX_FLAG_FROM_ADDRESS|GET_MODULE_HANDLE_EX_FLAG_UNCHANGED_REFCOUNT, reinterpret_cast<const char*>(address), &module)) {
-        /** @todo use FormatMessage() for a string error */
-        Error{} << "Utility::Directory::libraryLocation(): can't get library location:" << GetLastError();
+        WCHAR* errorStringW = nullptr;
+        FormatMessage(FORMAT_MESSAGE_FROM_SYSTEM|FORMAT_MESSAGE_ALLOCATE_BUFFER,
+            nullptr, GetLastError(), 0, reinterpret_cast<LPWSTR>(&errorStringW),
+            0, nullptr);
+        Containers::ScopeGuard e{errorStringW, LocalFree};
+        std::string errorStringUtf8 = Unicode::narrow(errorStringW);
+        /* The error message already contains a newline */
+        Error{Debug::Flag::NoNewlineAtTheEnd} <<
+            "Utility::Directory::libraryLocation(): can't get library location:" << errorStringUtf8.data();
```

Comment:
Hmm... maybe put this in a helper function in an unnamed namespace so it's not duplicated, or the `Implementation/WindowsError.h` we talked about?

Reply:
✅ (needs some cleanup, though, WIN32_LEAN_AND_MEAN etc)